.github/workflows/publish.yml

```diff
@@ -0,0 +1,38 @@
+name: publish
+
+on:
+  push:
+    paths:
+      - 'Bmx.CommandLine/**'
+      - 'Bmx.Core/**'
+      - 'Bmx.Idp.Okta/**'
+      - 'Bmx.Service.Aws/**'
+      - '.github/workflows/publish.yml'
+    branches:
+      - main
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        machine: [windows-latest, macos-latest, ubuntu-20.04]
+        architecture: [x64, arm64]
+        include:
+          - machine: windows-latest
+            platform: win
+          - machine: macos-latest
+            platform: osx
+          - machine: ubuntu-20.04
+            platform: linux
+    runs-on: ${{ matrix.machine }}
+    timeout-minutes: 5
+    steps:
+        - name: checkout
+          uses: Brightspace/third-party-actions@actions/checkout
+        - name: install prequisites
+          if: matrix.platform == "linux"
+          run: |
+            sudo apt-get install clang zlib1g-dev
+            sudo apk add clang build-base zlib-dev
+        - name: publish
+          run: dotnet publish -r ${{ matrix.platform }}-${{ matrix.architecture }} -c Release
```

Comment:
For linux arm 64 at the bottom of this section it mentions some extra work that I'm not totally understanding. https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/docs/compiling.md#cross-architecture-compilation

Reply:
I'm not sure if mac needs something special too. The page didn't mention it.